numba/core/typed_passes.py

```diff
@@ -437,16 +437,45 @@ def run_pass(self, state):
 
 
 @register_pass(mutates_CFG=False, analysis_only=True)
-class IRLegalization(AnalysisPass):
+class NoPythonSupportedFeatureValidation(AnalysisPass):
+    """NoPython Mode check: Validates the IR to ensure that features in use are
+    in a form that is supported"""
 
-    _name = "ir_legalization"
+    _name = "no_python_supported_feature_validation"
 
     def __init__(self):
         AnalysisPass.__init__(self)
 
     def run_pass(self, state):
         raise_on_unsupported_feature(state.func_ir, state.typemap)
         warn_deprecated(state.func_ir, state.typemap)
+        return False
+
+
+@register_pass(mutates_CFG=False, analysis_only=True)
+class ObjModeSupportedFeatureValidation(AnalysisPass):
+    """Object mode check: Validates the IR to ensure that features in use are
+    in a form that is supported"""
+
+    _name = "obj_mode_supported_feature_validation"
+
+    def __init__(self):
+        AnalysisPass.__init__(self)
+
+    def run_pass(self, state):
+        warn_deprecated(state.func_ir, state.typemap)
```

Comment:
I don't think `warn_deprecated` is needed either. All the deprecated feature are nopython features, and `typemap` is all `pyobject`.

Reply:
I think this is also likely true. Obj mode blocks can't return reflected containers and obj mode itself doesn't need to worry about it. Given this pass is now a NOP, remove it?

Reply:
yes remove it

Reply:
done in 8c8222a